parseAndPopulate/modulesComplicatedAlgorithms.py

```diff
@@ -133,9 +132,13 @@ def __init__(
         self.json_ytree = json_ytree
         self._trees: dict[str, dict[str, str]] = defaultdict(dict)
         self._unavailable_modules = []
+
         LOGGER.info('get all existing modules')
-        response = requests.get('{}/search/modules'.format(self._yangcatalog_api_prefix), headers=json_headers)
-        existing_modules = response.json().get('module', [])
+        existing_modules = fetch_modules(LOGGER)
+        if existing_modules is None:
+            LOGGER.error('module extraction from API has failed')
```

Comment:
This should be changed to something like:
```
Failed to fetch modules from API.
```
NOTE: Please change it also in all other files